src/components/PeopleManagement/EditGroupNameModal.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  ActionRow, Form, ModalDialog, Spinner, StatefulButton, useToggle,
+} from '@openedx/paragon';
+
+import LmsApiService from '../../data/services/LmsApiService';
+import GeneralErrorModal from './GeneralErrorModal';
+
+const EditGroupNameModal = ({
+  group, isOpen, close, setShowToast, setToastMessage,
+}) => {
+  const intl = useIntl();
+  const [isErrorOpen, openError, closeError] = useToggle(false);
+  const [name, setName] = useState(group?.name);
+  const [buttonState, setButtonState] = useState('default');
+  const editEnterpriseGroup = async () => {
+    setButtonState('pending');
+    const formData = { name };
+    const response = await LmsApiService.updateEnterpriseGroup(group?.uuid, formData);
+    if (response.status === 200) {
+      setButtonState('complete');
+      close();
+      setShowToast(true);
+      setToastMessage('Group name updated');
```

Comment:
in addition to these states, I think we should also trigger a re-render for the group name card. Currently the group name stays the same unless the user triggers a refresh.

https://github.com/user-attachments/assets/008be0a2-d055-4f6b-a3e4-4339e92378f7